core/Config.php

```diff
@@ -769,17 +770,37 @@ public function stats() {
 		return $this->miss ? ($this->hit / $this->miss) : 0;
 	}
 
+	/**
+	 * Return a cached value in the case of a hit, false otherwise.
+	 * For a more robust cache checking, use {@link checkAndGet()}
+	 *
+	 * @param  string $key The cache key
+	 * @return variant     Cached value, if hit. False otherwise
+	 */
 	public function get($key) {
+		list($hit, $result) = $this->checkAndGet($key);
+		return $hit ? $result : false;
+	}
+
+	/**
+	 * Checks for a cache hit and looks up the value by returning multiple values.
+	 * Distinguishes a cached 'false' value from a cache miss.
+	 *
+	 * @param  string $key The cache key
+	 * @return array  First element boolean, isHit. Second element the actual result.
+	 */
+	public function checkAndGet($key) {
 		if (isset($this->indexing[$key])) {
```

Comment:
Trying to read this code is hard; $i, $indexing etc aren't documented so it makes no sense to me. Worth documenting these if you ever get around to resurrecting this PR @sminnee :D